utils/datasets.py

```diff
@@ -347,19 +347,19 @@ def img2label_paths(img_paths):
         try:
             f = []  # image files
             for p in path if isinstance(path, list) else [path]:
-                p = str(Path(p))  # os-agnostic
-                parent = str(Path(p).parent) + os.sep
-                if os.path.isfile(p):  # file
+                p = Path(p)  # os-agnostic
+                if p.is_file():  # file
+                    parent = str(p.parent) + os.sep
                     with open(p, 'r') as t:
                         t = t.read().splitlines()
                         f += [x.replace('./', parent) if x.startswith('./') else x for x in t]  # local to global path
-                elif os.path.isdir(p):  # folder
-                    f += glob.iglob(p + os.sep + '*.*')
+                elif p.is_dir():  # dir
+                    f += glob.glob(str(p) + '/**/*.*', recursive=True)
```

Comment:
Something is wrong at this line. Program hangs on here.

Reply:
What OS are you on? Perhaps I need to make this OS robust. I will push an update here hold on.

Reply:
Ok I've pushed ec75dde which may help on Windows machines. If you are still having problems please try to reproduce with COCO128 (i.e. by copying one COCO128 into another) on a Google Colab notebook and share.